internal/pkg/publisher/publisher_test.go

```diff
@@ -118,7 +118,7 @@ func TestPublisher_EagerDrain(t *testing.T) {
 
 	pubWg := new(sync.WaitGroup)
 	timesync.Listen()
-	pub.Start(pubWg)
+	pub.Start(context.Background(), pubWg)
```

Comment:
Im aware this doesn't cause any issues today, but let's use `WithCancel` to properly close all goroutines in the tests you modified.

Reply:
Addressed in 01f7ecf